modules/gghealthd/src/health.c

```diff
@@ -249,8 +236,66 @@ GglError gghealthd_restart_component(GglBuffer component_name) {
     return GGL_ERR_OK;
 }
 
+static bool is_nucleus_component(GglBuffer component_name) {
+    if (ggl_buffer_has_prefix(component_name, GGL_STR("aws.greengrass."))) {
```

Comment:
This seems like it would give many false positives, many of our aws-provided generic components like aws.greengrass.StreamManager would match this.

It may be better to use the componentType info from the config to figure out what are the nucleus components vs not